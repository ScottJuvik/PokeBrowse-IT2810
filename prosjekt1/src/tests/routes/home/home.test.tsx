import Home from '@routes/home/home.tsx';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { render, screen, within } from '@testing-library/react';
import { RouterProvider, createMemoryRouter } from 'react-router-dom';
import { describe, expect, it } from 'vitest';

const queryClient = new QueryClient();

describe('Home component', () => {
    it("renders 'PokéBrowse' title on the home route", () => {
        const router = createMemoryRouter(
            [
                {
                    path: '/',
                    element: <Home />,
                },
            ],
            {
                initialEntries: ['/'],
            }
        );

        render(
            <QueryClientProvider client={queryClient}>
                <RouterProvider router={router} />
            </QueryClientProvider>
        );

        expect(within(screen.getByRole('heading', { level: 1 })).getByText(/Poké/i)).toBeInTheDocument();

        expect(within(screen.getByRole('heading', { level: 1 })).getByText(/Browse/i)).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Search for a Pokémon')).toBeInTheDocument();
        expect(screen.getByRole('button', { name: /Random/i })).toBeInTheDocument();
        expect(screen.getByRole('button', { name: /Pokémon/i })).toBeInTheDocument();
    });
});
