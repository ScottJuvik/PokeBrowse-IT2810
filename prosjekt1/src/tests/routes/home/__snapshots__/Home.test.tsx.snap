// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > matches the snapshot 1`] = `
<div>
  <div
    class="_homeContainer_aab42e"
  >
    <div
      class="_backgroundWrapper_aab42e"
    >
      <video
        autoplay=""
        class="_background_aab42e"
        id="backgroundVideo"
        loop=""
        preload="auto"
      >
        <source
          src="/videos/may_waterfall.mp4"
          type="video/mp4"
        />
      </video>
    </div>
    <main
      class="_content_aab42e"
    >
      <h1
        class="_title_aab42e"
      >
        Poké
        <span>
          Browse
        </span>
      </h1>
      <div
        class="_interactives_aab42e"
      >
        <div
          class="_inputWrapper_214055"
        >
          <form
            class="_searchWrapper_214055 _large_214055"
          >
            <input
              class="_searchInput_214055"
              placeholder="Search for a Pokémon"
              type="text"
              value=""
            />
            <button
              class="_searchButton_214055"
              type="submit"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.8 2.8L10.8 3.8 16.2 9.3 1 9.3 1 10.7 16.2 10.7 10.8 16.2 11.8 17.2 19 10z"
                />
              </svg>
            </button>
          </form>
        </div>
        <div
          class="_buttonWrapper_aab42e"
        >
          <button
            class="_random_aab42e"
            name="random"
            type="button"
          >
            Random
          </button>
          <button
            class="_pokemon_aab42e"
            name="pokemon"
            type="button"
          >
            Pokémon
          </button>
        </div>
      </div>
    </main>
  </div>
</div>
`;
